pipelines/batch/src/main/java/org/openmrs/analytics/JdbcFetchHapi.java

```diff
@@ -160,4 +165,34 @@ List<QueryParameterDescriptor> generateQueryParameters(
 
     return queryParameterList;
   }
+
+  /**
+   * Searches for the total number of resources for each resource type
+   *
+   * @param resourceList the resource types to be processed
+   * @param since the time from which the records need to be fetched
+   * @return a Map storing the counts of each resource type
+   */
+  public Map<String, Integer> searchResourceCounts(String resourceList, String since)
+      throws SQLException {
+    HashSet<String> resourceTypes = new HashSet<String>(Arrays.asList(resourceList.split(",")));
+    HashMap<String, Integer> hashMap = new HashMap<String, Integer>();
+    for (String resourceType : resourceTypes) {
+      StringBuilder builder = new StringBuilder();
+      builder.append("SELECT count(*) FROM hfj_resource res where res.res_type = ?");
+      if (since != null && !since.isBlank()) {
+        builder.append(" AND res.res_updated > '").append(since).append("'");
+      }
+      PreparedStatement statement =
```

Comment:
lets add comment for why we need TYPE_SCROLL_SENSITIVE and CONCUR_READ_ONLY

Reply:
This has been removed now